janus-idp/operator
# fix: Make sure to use backward-compatible names for resources created by the operator [RHIDP-2432]

## Description

This fixes a critical issue where, when upgrading existing resources using an operator upgraded from 1.1 to 1.2 (`main` branch), the operator would create and use a brand new database StatefulSet. This would cause the upgraded Backstage application to boot using an empty database, rather than starting with the existing database (which could already contain some data).

It also fixes a more general issue where resource naming was not backward-compatible, which would create for example 2 different ConfigMaps after an upgrade: one named `backstage-dynamic-plugins-<cr>` (created by the operator on the `main` branch) besides an already existing one named `<cr>-dynamic-plugins` (created in 1.1), which could create some confusion when looking at the resources in the namespace.

## Which issue(s) does this PR fix or relate to

- Fixes https://issues.redhat.com/browse/RHIDP-2432

## PR acceptance criteria


## How to test changes / Special notes to the reviewer

See the repro steps depicted in https://issues.redhat.com/browse/RHIDP-2432 and run the tests with the following command:
```
make test
make integration-test \
  ARGS=-focus='creates runtime object using raw configuration then updates StatefulSet to replace some immutable fields' \
  USE_EXISTING_CLUSTER=true
```
